src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -210,6 +217,11 @@ private DeclarationModifiers CreateMethodModifiers()
                 var isStatic = !this.AnalyzerResult.UseInstanceMember;
                 var isReadOnly = this.AnalyzerResult.ShouldBeReadOnly;
 
+                if (LocalFunction && !this.Options.GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction).Value)
+                {
+                    isStatic = false;
+                }
```

Comment:
do you need to check language version as well?

Reply:
LocalFunction should already be false if the language version doesn't support it right?

Reply:
doi, static local is c# 8 only

Reply:
although, logistically I'm not opposed to the MethodGenerator caring if the language version supports local static. This is just stating that the user prefers it to be static. Having the check where it's at in `GenerateModifiers` makes sense to me

Reply:
I decided to switch the check over to CSharpCodeGenerator since, for example, if the user has a camelCase naming rule that applies exclusively to static local functions but they're in an earlier version of C#, the naming rule shouldn't apply. I think this is an extreme edge case but thought it probably wouldn't hurt to just move the check over.